webapp/src/Controller/Jury/ClarificationController.php

```diff
@@ -346,6 +346,25 @@ public function changeQueueAction(Request $request, int $clarId): Response
         return $this->redirectToRoute('jury_clarification', ['id' => $clarId]);
     }
 
+    #[Route('/check-claimed', name: 'check_if_claimed', methods: ['GET'])]
```

Comment:
Instead of adding another endpoint here, let's just re-use the existing API endpoint for clarifications.

It currently doesn't have the info on whether a clarification has been claimed, but that can be added easily.

Reply:
When we do this, we also have the info on whether it has been answered before and can use that as well.

Reply:
In which API should this be added?

Reply:
I am talking about the default API clarification controller:

```
http http://localhost/domjudge/api/contests/demo/clarifications/1
HTTP/1.1 200 OK
Cache-Control: max-age=0, must-revalidate, private
Connection: Keep-Alive
Content-Security-Policy: default-src 'self'; style-src 'self' 'unsafe-inline'; script-src 'self' 'unsafe-inline' 'unsafe-eval'; img-src 'self' data:
Content-Type: application/json
Date: Fri, 01 Nov 2024 17:01:08 GMT
Expires: Fri, 01 Nov 2024 17:01:08 GMT
Keep-Alive: timeout=5, max=100
Referrer-Policy: same-origin
Server: Apache/2.4.62 (Debian)
Transfer-Encoding: chunked
X-Content-Type-Options: nosniff
X-Current-Contest: 
X-Debug-Token: 55677e
X-Debug-Token-Link: http://localhost/domjudge/_profiler/55677e
X-Frame-Options: DENY
X-Robots-Tag: noindex
X-XSS-Protection: 1; mode=block

{
    "answered": false,
    "clarid": 1,
    "contest_time": "233:54:09.605",
    "from_team_id": "domjudge",
    "id": "1",
    "problem_id": "boolfind",
    "reply_to_id": null,
    "text": "fsdfsdfdf",
    "time": "2024-11-01T10:47:16.605+00:00",
    "to_team_id": null
}
```

As you can see, it doesn't output `jury_member` as of now. The implementation of it is really simple: https://github.com/DOMjudge/domjudge/blob/main/webapp/src/Controller/API/ClarificationController.php#L81

That means all the logic is in the entity definition. If we check there, you can see the following: https://github.com/DOMjudge/domjudge/blob/main/webapp/src/Entity/Clarification.php#L67

This means it is excluded from serialization in the API. We can change it to
```
 #[Serializer\Groups([ARC::GROUP_RESTRICTED_NONSTRICT])]
 ```
 and it should appear.

Reply:
@meisterT 

How can I know the cid on the frontend?

The API should look like this:

**/api/v4/contests/{cid}/clarifications/{id}**

And do I need to call this API directly on the frontend using fetch?